importer/src/main/java/org/hiero/mirror/importer/migration/ExpandBalancePartitionsMigration.java

```diff
@@ -0,0 +1,271 @@
+            Environment environment) {
+
+        super(importerProperties.getMigration(), jdbcOperationsProvider, dbProperties.getSchema());
+        this.treasuryAccountId = systemEntity.treasuryAccount().getId();
+        this.v2 = environment.acceptsProfiles(Profiles.of("v2"));
+        this.timePartitionServiceProvider = timePartitionServiceProvider;
+    }
+
+    @Override
+    protected boolean performSynchronousSteps() {
+        this.partitionIndex.set(0);
+        this.accountBalancePartitions =
+                timePartitionServiceProvider.getObject().getTimePartitions(NEW_ACCOUNT_BALANCE_TABLE).stream()
+                        .sorted(Comparator.comparing((TimePartition p) ->
+                                        p.getTimestampRange().upperEndpoint())
+                                .reversed())
+                        .toList();
+        log.info("Performing synchronous steps for balance migration");
+
+        Boolean performAsync = getTransactionOperations().execute(status -> {
+            final var maxBalanceFileTimestamp =
+                    Objects.requireNonNull(getJdbcOperations().queryForObject(LAST_BALANCE_FILE, Long.class));
+
+            if (maxBalanceFileTimestamp == -1L) {
+                log.info("No balance files found, skipping migration");
+                return Boolean.FALSE;
+            }
+
+            final var sixHoursNs = 6L * 60 * 60 * 1_000_000_000L;
+            final var minTimestamp = Math.max(0L, maxBalanceFileTimestamp - sixHoursNs);
+
+            log.info(
+                    "Synchronously backfilling partitions overlapping [{}, {}]", minTimestamp, maxBalanceFileTimestamp);
+
+            while (partitionIndex.get() < accountBalancePartitions.size()) {
+                final var idx = partitionIndex.get();
+                final var partition = accountBalancePartitions.get(idx);
+                final var range = partition.getTimestampRange();
+                final var from = range.lowerEndpoint();
+                final var to = range.upperEndpoint();
+
+                if (to <= minTimestamp) {
+                    log.info("Reached partition {} no more synchronous partitions to process", range);
+                    break;
+                }
+
+                final var intersects = (from <= maxBalanceFileTimestamp);
+
+                if (intersects) {
+                    log.info("Synchronously processing partition {}", range);
+                    processPartition(partition);
+                }
+
+                partitionIndex.incrementAndGet();
+            }
+
+            return Boolean.TRUE;
+        });
+
+        if (BooleanUtils.isNotTrue(performAsync) || partitionIndex.get() >= accountBalancePartitions.size()) {
+            finalizeMigration();
+            return false;
+        }
+
+        return true;
+    }
+
+    @Override
+    protected TimePartition getInitial() {
+        return accountBalancePartitions.get(partitionIndex.getAndIncrement());
+    }
```

Comment:
can simplify it to just return the index to the time partitions list

```suggestion
    protected Integer getInitial() {
        return 0;
    }
```